lib/std/testing.zig

```diff
@@ -21,6 +21,57 @@ pub var base_allocator_instance = std.heap.FixedBufferAllocator.init("");
 /// TODO https://github.com/ziglang/zig/issues/5738
 pub var log_level = std.log.Level.warn;
 
+/// Non-zero counts imply test failure. The value is reset for every test.
+pub var error_count: usize = 0;
+
+/// Fail the test with a formatted message but continue execution, unlike fatal.
+/// The common pattern is as follows.
+///
+/// ```zig
+///     if (got != want)
+///         testing.fail("got {}, want {}", got, want);
+/// ```
```

Comment:
Markdown code blocks should be avoided in favor of decltests:
```zig
test fail {
     ...
     if (got != want)
        testing.fail("got {}, want {}", got, want);
     ...
}
```

Reply:
I meant to remove it from the doc comment like [the `test addOne` in the langref](https://ziglang.org/documentation/master/#Zig-Test).

Reply:
Does such test code end up in the documentation somehow? Or do we expect people to read tests on top of doc comments?

Reply:
I could try and put the convention into words, if that's what's prefered?

Reply:
> Does such test code end up in the documentation somehow?

Yes, see https://ziglang.org/documentation/master/std/#A;std:enums.tagName